### DISCUSSION TITLE
Is it possible to open the selected file in the list of coc-references in a vertical split?

### DISCUSSION THREAD

**Discussion by harvey-wei** [neoclide/coc.nvim#3888] [Q&A] [answered]
Note: Without changing the coc.preferences.jumpCommand?

**Comment by chemzqm** [comment, marked as answer]
Read the doc `:h CocAction('jumpReferences')`

**Reply by harvey-wei** [reply]
Hello!
Thanks! But I don't know how to set in init.vim. "jumpReferences" [{openCommand}]

**Reply by harvey-wei** [reply]
> Read the doc `:h CocAction('jumpReferences')`

nmap <silent> gr :call CocAction('jumpReferences, 'vsplit')<CR> did not work as expected!

**Reply by chemzqm** [reply]
There is `vsplit` action for locations list.

**Reply by harvey-wei** [reply]
> There is `vsplit` action for locations list.

How to set it?

**Reply by chemzqm** [reply]
Use `<tab>` or create your own key-mapping for do action.

**Reply by ravicious** [reply]
How do you create key-mappings for do actions? The readme mentions "do action" only once and not in the context of making custom mappings.

**Reply by ssh352** [reply]
> > Read the doc `:h CocAction('jumpReferences')`
> 
> nmap gr :call CocAction('jumpReferences, 'vsplit') did not work as expected!

same issue here, how to view references in a vertical or in a new tab?

**Reply by ssh352** [reply]
> How do you create key-mappings for do actions? The readme mentions "do action" only once and not in the context of making custom mappings.

Did you figure out how to do it?

**Reply by fannheyward** [reply]
@ssh352 if there's only one references location, `CocAction('jumpReferences, 'vsplit')` will jump to references with `vsplit`, otherwise, all references will be displayed in a list view.

**Reply by ssh352** [reply]
> @ssh352 if there's only one references location, `CocAction('jumpReferences, 'vsplit')` will jump to references with `vsplit`, otherwise, all references will be displayed in a list view.

Thanks, but most of the time there are multiple references. A horizontal split looks too congested on a wide and flat monitor.
